src/Microsoft.Graph.Cli.Core/Authentication/AuthenticationServiceFactory.cs

```diff
@@ -13,33 +14,40 @@ public class AuthenticationServiceFactory
 {
     private readonly IPathUtility pathUtility;
 
-    public AuthenticationServiceFactory(IPathUtility pathUtility)
+    private readonly AuthenticationOptions? authenticationOptions;
+
+    public AuthenticationServiceFactory(IPathUtility pathUtility, AuthenticationOptions? authOptions)
     {
         this.pathUtility = pathUtility;
+        this.authenticationOptions = authOptions;
     }
 
-    public virtual async Task<ILoginService> GetAuthenticationServiceAsync(AuthenticationStrategy strategy, string? tenantId, string? clientId, CancellationToken cancellationToken = default)
+    public virtual async Task<ILoginService> GetAuthenticationServiceAsync(AuthenticationStrategy strategy, string? tenantId, string? clientId, string? certificateName, string? certificatePath, string? certificateThumbPrint, CancellationToken cancellationToken = default)
     {
         switch (strategy)
         {
             case AuthenticationStrategy.DeviceCode:
                 return await GetDeviceCodeLoginServiceAsync(tenantId, clientId, cancellationToken);
             case AuthenticationStrategy.InteractiveBrowser:
                 return await GetInteractiveBrowserLoginServiceAsync(tenantId, clientId, cancellationToken);
+            case AuthenticationStrategy.ClientCertificate:
```

Comment:
quick question: what if the certificate is password protected?

Reply:
In the test, it fails to open the certificate file. There doesn't seem to be an API to provide a password in the credential class. Should we remove the path option for the moment? PowerShell doesn't have an option to provide a path.

Reply:
I believe the main reason why they don't have a path or password parameter is because of the nature of PowerShell and the fact they can pass the x509Certificate object directly. This way people can rely on other commandlets (outside of Graph, native to PS) to handle reading the certificate, using a password etc...
I believe the password needs to be passed to the store, or the certificate when you try to load it from a path (at least that's how it works in Java/Go) and not to the credentials from azure identity.

Reply:
Okay. I think we need to discuss the security implications of this option. For example, we need to store the password for subsequent use. This is because the certificate is required to initialize the credential object.

Reply:
@peombwa what are you doing in PS when a certificate is used? from a vulnerability perspective storing a certificate with the password protected key + key to access it or storing certificate with unprotected key is similar.

Reply:
On the CLI, if the user provides a certificate file, they're asked to provide a password. They can leave it blank if there's no password.